Лаба 3 ООП/Лабораторная 3.py

```diff
@@ -0,0 +1,81 @@
+class Book:
+    """ Базовый класс книги. """
+    def __init__(self, name: str, author: str):
+        self._name = None   # делаем атрибут защищенным, добавив нижнее подчеркивание
+        self.set_name(name)    # устанавливаем с помощью метода название книги
+        self._author = None  # делаем атрибут защищенным, добавив нижнее подчеркивание
+        self.set_author(author)  # устанавливаем с помощью метода автора
+
+    # Публичный метод, который внутри работает с защищенным атрибутом self._name
+    def get_name(self) -> int:
+        return self._name
+
+    # Публичный метод, который внутри работает с защищенным атрибутом self._name
+    def set_name(self, name: str) -> None:
+        if not isinstance(name, str):
+            raise TypeError("Название должно быть типа str")
+        self._name = name
+
+    # Публичный метод, который внутри работает с защищенным атрибутом self._author
+    def get_author(self) -> int:
+        return self._author
+
+    # Публичный метод, который внутри работает с защищенным атрибутом self._author
+    def set_author(self, author: str) -> None:
+        if not isinstance(author, str):
+            raise TypeError("Автор должен быть типа str")
+        self._author = author
+
+    def __str__(self):
+        return f"Книга {self.name}. Автор {self.author}"
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}(name={self.name!r}, author={self.author!r})"
+
+
+class PaperBook(Book):
+    def __init__(self, name: str, author: str, pages: int):
+        super().__init__(name, author)
+        self.pages = pages
+
+    @property
+    def pages(self) -> int:
+        return self.pages

```

Comment:
```suggestion
        return self._pages
```
Если возвращать переменную без нижнего подчеркивания, то метод уйдет в рекурсию